witness/golang/omniwitness/usbarmory/internal/storage/slots/journal.go

```diff
@@ -43,11 +43,15 @@ type BlockReaderWriter interface {
 	WriteBlocks(lba uint, b []byte) error
 }
 
+// Hasher returns a hash of the data available from the given reader.
+type Hasher func(io.Reader) ([32]byte, error)
```

Comment:
Any reason to take an `io.Reader` ? It looks like there's always the `[]byte` read into memory when we do the hashing.